src/components/AudacitySetup.tsx

```diff
@@ -0,0 +1,218 @@
+import React from 'react';
+import { useGlobal } from 'reactn';
+import { connect } from 'react-redux';
+import { IState, IAudacitySetupStrings } from '../model';
+import localStrings from '../selector/localize';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+  Button,
+  IconButton,
+  Avatar,
+  List,
+  ListItem,
+  ListItemAvatar,
+  ListItemText,
+  ListItemSecondaryAction,
+  DialogTitle,
+  Dialog,
+  DialogActions,
+} from '@material-ui/core';
+import CheckIcon from '@material-ui/icons/Check';
+import RefreshIcon from '@material-ui/icons/Refresh';
+import AudacityLogo from '../control/AudacityLogo';
+import {
+  hasAudacity,
+  hasAuacityScripts,
+  hasPython,
+  enableAudacityScripts,
+} from '../utils';
+
+const useStyles = makeStyles({
+  root: {
+    '& .MuiDialog-paper': {
+      minWidth: '450px',
+    },
+  },
+  title: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+  },
+  avatar: {
+    color: 'green',
+  },
+});
+
+enum Step {
+  Audacity,
+  Scripting,
+  Python,
+}
+
+interface StepType {
+  item: Step;
+  choice: string;
+  action: string;
+  url: string | (() => Promise<void>);
+  help?: boolean;
+  testIt: () => Promise<boolean>;
+}
+
+interface IStateProps {
+  t: IAudacitySetupStrings;
+}
+
+export interface AudacitySetupProps extends IStateProps {
+  open: boolean;
+  onClose: () => void;
+}
+
+function AudacitySetup(props: AudacitySetupProps) {
+  const { t } = props;
+  const classes = useStyles();
+  const { onClose, open } = props;
+  const [allAudacity, setAllAdudacity] = useGlobal('allAudacity');
+  const [satisfied, setSatisfied] = React.useState<boolean[]>([
+    false,
+    false,
+    false,
+  ]);
+  const steps: StepType[] = [
+    {
+      item: Step.Audacity,
+      choice: t.audacityInstalled,
+      action: t.getInstaller,
+      url: 'https://www.audacityteam.org/download/',
+      testIt: hasAudacity,
+    },
+    {
+      item: Step.Scripting,
+      choice: t.scriptingEnabled,
+      action: t.enable,
+      url: enableAudacityScripts,
+      help: true,
+      testIt: hasAuacityScripts,
+    },
+    {
+      item: Step.Python,
+      choice: t.pythonInstalled,
+      action: t.getInstaller,
+      url: 'https://www.python.org/downloads/',
+      testIt: hasPython,
+    },
+  ];
+
+  const handleClose = () => {
+    onClose();
+  };
+
+  const doEval = async () => {
+    hasAudacity().then((r) => setSatisfied((a) => [r, a[1], a[2]]));
```

Comment:
wondering about async nature of setSatisfied.  You may get results from one overwriting results from another.   
you might need to build up one result (using promises.all?) and do one setSatisfied

Reply:
One of the reasons I didn't do this is because it was throwing an error when the registry key for an item wasn't installed but I have trapped the error now so I think this idea will work. It works in my test.